### PR TITLE
Add initial `rust-toolchain.toml` schema

### PR DESCRIPTION
**Pull request by ian-h-chamberlain**
First part of #228.

I am hoping to get to `.cargo/config.toml` next, but it has _lots_ of fields so may take a bit longer.

Please let me know if there's anything you'd like added or changed, I think I got most of the docs (pulled from [the rustup book](https://rust-lang.github.io/rustup/index.html)), but may have overlooked or typoed something.

### CODE REVIEW

File: schemas/rust-toolchain.toml.json [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+                {
+                  "$comment": "Empty schema allows for custom rustup toolchains, specific nightlies, etc."
+                }
+              ],
+              "x-taplo": {
+                "links": {
+                  "key": "https://rust-lang.github.io/rustup/concepts/channels.html"
+                },
+                "docs": {
+                  "main": "Rust is released to three different \"channels\": stable, beta,\nand nightly."
```
Comment (ian-h-chamberlain) [inline schemas/rust-toolchain.toml.json:41]: There were a couple instances where I wasn't sure if I should put the upstream docs into `x-taplo.docs.main` or just `description`, is there a preference on this?

Comment (tamasfe) [reply schemas/rust-toolchain.toml.json:41]: I'd say avoid taplo-specific extensions whenever possible, so prefer `description` in this case. The main reason why `x-taplo.docs.main` exists is that `x-taplo` is carried over `$ref`s in the current implementation, so it allows overriding the docs for referenced schemas, but I'm revisiting this decision in the next version as it complicates things.

**Comment by tamasfe** [comment]
Thanks, looks good to me!

> Please let me know if there's anything you'd like added or changed, I think I got most of the docs (pulled from [the rustup book](https://rust-lang.github.io/rustup/index.html)), but may have overlooked or typoed something

Ufortunately I don't really have time so I'd rather not review the schema itself. So it's up to you when you think it's ready to be merged, it can always be refined later as well.

**Comment by ian-h-chamberlain** [comment]
> Thanks, looks good to me!
> 
> > Please let me know if there's anything you'd like added or changed, I think I got most of the docs (pulled from [the rustup book](https://rust-lang.github.io/rustup/index.html)), but may have overlooked or typoed something
> 
> Ufortunately I don't really have time so I'd rather not review the schema itself. So it's up to you when you think it's ready to be merged, it can always be refined later as well.

Understood, I will address your comment about `x-taplo` docs and give it a once-over. Btw, it looks like there was a CI failure, but it seems unrelated to my change? Just want to verify in case I need to push some change to pass CI.
Related note:

**Comment by tamasfe** [comment]
> Btw, it looks like there was a CI failure, but it seems unrelated to my change? Just want to verify in case I need to push some change to pass CI.

No, sorry, the CI has been broken for a while, I'm trying to push the new version out today and fix it along with it.

> Related note: are there any test cases I should add to actually use the new schema? Even if it's just a basic toolchain file to make sure the schema isn't broken (I've only done manual testing thus far).

The schema should be verified by a JSON schema validator, but other than that there aren't any more specific tests yet.

**Comment by ian-h-chamberlain** [comment]
Ok, I think this should be good to go now. Merge at your convenience I guess?

**Pull request merged**